src/main/scala/mesosphere/marathon/state/Residency.scala

```diff
@@ -0,0 +1,16 @@
+package mesosphere.marathon.state
+
+import com.wix.accord.dsl._
+import mesosphere.marathon.Protos.ResidencyDefinition.TaskLostBehavior
+
+case class Residency(relaunchEscalationTimeoutSeconds: Long, taskLostBehavior: TaskLostBehavior)
```

Comment:
The actual idea behind as discussed was to use `relaunchEscalationTimeoutSeconds` for all terminal tasks (FAILED, FINISHED, etc) and to have the `taskLostBehavior` define the behavior for lost tasks.
That means that
- if a task fails, Marathon will wait for an offer matching its reservation labels and volumes for the amount of time defined by `relaunchEscalationTimeoutSeconds`. When the timeout is reached, it will relaunch the task on another agent.
- if a task is lost, Marathon will either do nothing (`WAIT_FOREVER`), or apply the same logic as above (although it most probably won't get an offer because the node is gone).

It might make sense to immediately relaunch elsewhere upon `TASK_LOST`, and to apply a `taskTerminationBehavior` instead to all kinds of terminal task states, respectively. All based on assumption though …